modules/apps/portal-crypto/portal-crypto-hash-generator-api/src/main/java/com/liferay/portal/crypto/hash/generator/HashGenerator.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.crypto.hash.generator;
+
+import org.osgi.annotation.versioning.ConsumerType;
+
+/**
+ * @author Arthur Chan
+ */
+@ConsumerType
+public interface HashGenerator {
```

Comment:
we need to move this interface to the `-spi` module since it is an extension point now.

Reply:
Right, will do